testing/test_workermanage.py

```diff
@@ -82,11 +82,15 @@ def test_popen_makegateway_events(
         call = hookrecorder.popcall("pytest_xdist_setupnodes")
         assert len(call.specs) == 2
 
-        call = hookrecorder.popcall("pytest_xdist_newgateway")
-        assert call.gateway.spec == execnet.XSpec("execmodel=main_thread_only//popen")
-        assert call.gateway.id == "gw0"
-        call = hookrecorder.popcall("pytest_xdist_newgateway")
-        assert call.gateway.id == "gw1"
+        # check expected gateways
+        gw_calls = [
+            hookrecorder.popcall("pytest_xdist_newgateway"),
+            hookrecorder.popcall("pytest_xdist_newgateway"),
```

Comment:
We can also sort the `pytest_xdist_newgateway` calls inside setup_nodes to maintain behavior